examples/release_highlights/plot_release_highlights_0_24_0.py

```diff
@@ -0,0 +1,251 @@
+#
+# The plot shows that the new native support for categorical features leads
+# to models that are almost as fast to train as model trained on arbitrarily
+# integer-encoded categorical values, while being more expressive than both
+# one-hot encoding and ordinal encoding. However, to use the new
+# `categorical_features` parameter, it is still required to preprocess the
+# data within a pipeline as demonstrated in this :ref:`example <sphx_glr_auto_examples_ensemble_plot_gradient_boosting_categorical.py>`.
+
+##############################################################################
+# Improved performances of HistGradientBoosting estimators
+# --------------------------------------------------------
+# The memory footprint of :class:`ensemble.HistGradientBoostingRegressor` and
+# :class:`ensemble.HistGradientBoostingClassifier` has been significantly
+# improved during calls to `fit`. In addition, histogram initialization is now
+# done in parallel which results in slight speed improvements.
+# See more in the `Benchmark page
+# <https://scikit-learn.org/scikit-learn-benchmarks/>`_.
+
+##############################################################################
+# New self-training meta-estimator
+# --------------------------------
+# A new self-training implementation, based on `Yarowski's algorithm
+# <https://doi.org/10.3115/981658.981684>`_ can now be used with any
+# classifier that implements :term:`predict_proba`. The sub-classifier
+# will behave as a
+# semi-supervised classifier, allowing it to learn from unlabeled data.
+# Read more in the :ref:`User guide <self_training>`.
+
+import numpy as np
+from sklearn import datasets
+from sklearn.semi_supervised import SelfTrainingClassifier
+from sklearn.svm import SVC
+rng = np.random.RandomState(42)
+iris = datasets.load_iris()
+random_unlabeled_points = rng.rand(iris.target.shape[0]) < 0.3
+iris.target[random_unlabeled_points] = -1
+svc = SVC(probability=True, gamma="auto")
+self_training_model = SelfTrainingClassifier(svc)
+self_training_model.fit(iris.data, iris.target)
+
+##############################################################################
+# New SequentialFeatureSelector transformer
+# -----------------------------------------
+# A new iterative transformer to select features is available:
+# :class:`~sklearn.feature_selection.SequentialFeatureSelector`.
+# Sequential Feature Selection can add features one at a time (forward
+# selection) or remove features from the list of the available features
+# (backward selection), based on a cross-validated score maximization.
+# See the :ref:`User Guide <sequential_feature_selection>`.
+
+from sklearn.feature_selection import SequentialFeatureSelector
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.datasets import load_iris
+
+X, y = load_iris(return_X_y=True, as_frame=True)
+feature_names = X.columns
+knn = KNeighborsClassifier(n_neighbors=3)
+sfs = SequentialFeatureSelector(knn, n_features_to_select=2)
```

Comment:
@NicolasHug this comment is not related to this PR, but I was surprised when reading it that there is no way to automatically select the number of samples based on a stop of improvements in the cross-validated score.

Also, it would be interesting for model inspect to record the ordering of the selected features along with the cross-validation scores at each step of the process. This would give an interesting alternative to permutation feature importance.

Reply:
Since the original PR had been open for years, I tried to implement the simplest possible version so that reviews would be easy.

That form of early-stopping is definitely something we can consider. We don't have it for RFE, where it might be relevant too (although we have RFECV).